src/libraries/EthFlowOrder.sol

```diff
@@ -47,6 +47,10 @@ library EthFlowOrder {
     /// @dev An order that is owned by this address is an order that has not yet been assigned.
     address public constant NO_OWNER = address(0);
 
+    /// @dev An order that is owned by this address is an order that has been invalidated. Note that this address cannot
+    /// be directly used to create orders.
+    address public constant INVALIDATED_OWNER = address(type(uint160).max);
```

Comment:
Why don't you set it to NO_OWNER? 

Is it because the order is still existing in the settlement contract and hence a new order placement would fail? Probably deleting the old order is too expensive?

Reply:
The worse that could happen is that the order was partially filled, the user cancels it, and then another completely different user creates an order with the same parameters. Then they would "inherit" the state of `filledAmount`, which would mean that (part of) their ETH is lost forever. (The same would happen for a fill-or-kill order that was cancelled, which is currently possible and returns 0 ETH.)
It's not possible to just clear the state of `filledAmount`, so I don't see a way around this. The most similar function is `invalidateOrder` in the settlement contract, but this just sets `filledAmount` to maxUint256.